openmc/filter.py

```diff
@@ -1875,6 +1875,9 @@ class EnergyFunctionFilter(Filter):
         A grid of energy values in [eV]
     y : iterable of Real
         A grid of interpolant values in [eV]
+    interpolation : str
+        Interpolation scheme: {'histogram', 'linear-linear', 'linear-log',
+        'log-linear', 'log-log', 'quadratic', 'cubic'}
```

Comment:
I was originally thinking something like this might be useful. However I agree with @pshriwise and this sort of doc comment is not very generic. I guess dose is quite a popular use of the EnergyFilter but perhaps I can find somewhere else to include this sort of thing in the future.

```suggestion
    interpolation : str
        Interpolation scheme: {'histogram', 'linear-linear', 'linear-log',
        'log-linear', 'log-log', 'quadratic', 'cubic'}. One example reason
        for changing the interpolation schemes is when simulating
        absorbed dose or effective dose. IRCP recommend using a
        'cubic' Interpolation scheme for dose calculations. See
        https://journals.sagepub.com/doi/pdf/10.1177/ANIB_40_2-5
        for more details.
```

Reply:
Yeah, I think that's a little too narrow for this filter, which could be used for many different things. Perhaps instead, we can have an example Jupyter notebook showing a dose calculation with the recommended interpolation.